source/frontend/src/features/mapSideBar/shared/header/HistoricalNumberFieldView.tsx

```diff
@@ -0,0 +1,85 @@
+import { useMemo } from 'react';
+import styled from 'styled-components';
+
+import ExpandableTextList from '@/components/common/ExpandableTextList';
+import { Dictionary } from '@/interfaces/Dictionary';
+import { ApiGen_Base_CodeType } from '@/models/api/generated/ApiGen_Base_CodeType';
+import { ApiGen_CodeTypes_HistoricalFileNumberTypes } from '@/models/api/generated/ApiGen_CodeTypes_HistoricalFileNumberTypes';
+import { ApiGen_Concepts_HistoricalFileNumber } from '@/models/api/generated/ApiGen_Concepts_HistoricalFileNumber';
+import { exists } from '@/utils';
+
+export interface IHistoricalNumbersViewProps {
+  historicalNumbers: ApiGen_Concepts_HistoricalFileNumber[];
+}
+
+interface HistoricalGroup {
+  historicalType: ApiGen_Base_CodeType<string>;
+  historicalValues: Dictionary<ApiGen_Concepts_HistoricalFileNumber>;
+  otherDescription: string;
+  propertyKey: string;
+}
+
+export const HistoricalNumberFieldView: React.FC<IHistoricalNumbersViewProps> = ({
+  historicalNumbers,
+}) => {
+  const uniqueNumbers = useMemo(() => {
+    const flatNumberDictionary: Dictionary<HistoricalGroup> = historicalNumbers
+      .filter(exists)
+      .reduce((accumulator, h) => {
+        const historicalTypeKey = `[${h.historicalFileNumberTypeCode.id}-${h.otherHistFileNumberTypeCode}]`;
+        if (!(historicalTypeKey in accumulator)) {
+          accumulator[historicalTypeKey] = {
+            historicalType: h.historicalFileNumberTypeCode,
+            historicalValues: {},
+            otherDescription: h.otherHistFileNumberTypeCode,
+            key: '',
+          };
+        }
+
+        const historicalValueKey = `[${h.id}][${h.historicalFileNumberTypeCode.id}][${h.otherHistFileNumberTypeCode}][${h.historicalFileNumber}]`;
+        accumulator[historicalTypeKey].historicalValues[historicalValueKey] = h;
```

Comment:
I think this just needs to check if the value is already in this key before adding.